packages/react/src/TooltipV2/Tooltip.tsx

```diff
@@ -213,6 +214,10 @@ export const Tooltip = React.forwardRef(
 
     const [isPopoverOpen, setIsPopoverOpen] = useState(false)
 
+    const timeoutRef = React.useRef(0)
```

Comment:
I didn't really realize how awkward this was with cancellation since you still need to keep track of the timeout 🤔 I think the type for this should be `number | null` technically and then we set it to `null` after clearing it but at that point the hook isn't doing much for us is it 😅 (really just has the effect clean-up)